File: QuanLyShop_Java/QuanLyShop/src/QuanLyNhanVien.java
import java.util.ArrayList;
import java.util.Collection;
import java.util.List;
import java.util.Scanner;

public class QuanLyNhanVien {
	// khởi tạo,thêm, sửa, xóa, sắp xếp, hiển thị danh sách nhân viên
	public static Scanner sc = new Scanner(System.in);
	public static List<NhanVien> lNhanVien;

	public static List<NhanVien> getlNhanVien() {
		return lNhanVien;
	}

	protected void setlNhanVien(List<NhanVien> NhanViens) {
		QuanLyNhanVien.lNhanVien = NhanViens;
	}

	public QuanLyNhanVien()
	{

	}

	public void KhoiTaoNhanVien() {
		lNhanVien = new ArrayList<NhanVien>();
		// String MaNV, String Ten, int NamSinh, String SDT, String DiaChi, String CMND,
		// int SoNgayLam, int GioTangCa, String LoaiNV

		// quản lý
		NhanVien ql1 = new QuanLy("QL01", "Trần Thị Quản Lý", 1999, "0988188449", "Quận 9, Tp.HCM", "251274983", 0, 0,
				"QL", "Nữ");
		lNhanVien.add(ql1);
		NhanVien ql2 = new QuanLy("QL02", "Nguyễn Trần Quản Lý", 1999, "0988188733", "Quận 8, Tp.HCM", "251287483", 0,
				0, "QL", "Nam");
		lNhanVien.add(ql2);

		// thu ngân
		NhanVien tn1 = new ThuNgan("TN01", "Lê Quỳnh Thu Ngân", 1998, "0986978673", "Quận Thủ Đức, Tp.HCM", "251274453",
				0, 0, "TV", "Nữ");
		lNhanVien.add(tn1);
		NhanVien tn2 = new ThuNgan("TN02", "Cao Thu Ngân", 1997, "0984567673", "Quận Bình Thạnh, Tp.HCM", "256784453",
				0, 0, "TV", "Nam");
		lNhanVien.add(tn2);

		// tư vấn
		NhanVien tv1 = new TuVan("TV01", "Cao Mỹ Tư Vấn", 2001, "0987983245", "Quận 2, Tp.HCM", "245274983", 0, 0, "TV",
				"Nữ");
		lNhanVien.add(tv1);
		NhanVien tv2 = new TuVan("TV02", "Trần Nguyễn Tư Vấn", 2001, "0935983245", "Quận 3, Tp.HCM", "246274983", 0, 0,
				"TV", "Nam");
		lNhanVien.add(tv2);

	}

	public void showNhanVien()
	{
		Scanner sc = new Scanner(System.in);
		System.out.println("____Chọn danh sách: ");
		System.out.println("	0.Toàn bộ nhân viên");
		System.out.println("	1.Quản Lý");
		System.out.println("	2.Thu Ngân");
		System.out.println("	3.Tư Vấn Mua Hàng");
		int ch = 0;
		ch = Integer.parseInt(sc.next());
		String l = null;
		if (ch == 1)
			l = "QL";
		if (ch == 2)
			l = "TN";
		if (ch == 3)
			l = "TV";
		switch (ch) {
		case 0:
			showAllNhanVien();
			break;
		case 1:
			showQuanLy();
			break;
		case 2:
			showThuNgan();
			break;
		case 3:
			showTuVan();
			break;
		default:
			break;
		}
	}

	public void showAllNhanVien()
	{
		System.out.println("                               ***DANH SÁCH NHÂN VIÊN FASHION SHOP***");
		System.out.println(
				"    ____________________________________________________________________________________________________________________");
		System.out.println(
				"    | STT |Mã NV|       Họ và Tên       |Giới Tính|    CMND    |Năm sinh|             Địa chỉ            |Số Điện Thoại|");
		System.out.println(
				"    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
		int i = 1;
		for (NhanVien nv : lNhanVien)
		{
			System.out.printf("    |%3d  |%5s|%23s|%7s  |%12s|%8d|%32s|%13" + "s|\n", i, nv.sMaNV, nv.sTen,
					nv.sGioiTinh, nv.sCMND, nv.iNamSinh, nv.sDiaChi, nv.sSDT);
			i++;
		}
		System.out.println(
				"    ____________________________________________________________________________________________________________________");
		System.out.println();
	}

	public void showQuanLy()
	{
		System.out.println("                            ***DANH SÁCH NHÂN VIÊN THU NGÂN FASHION SHOP***");
		System.out.println(
				"    ____________________________________________________________________________________________________________________");
		System.out.println(
				"    | STT |Mã NV|       Họ và Tên       |Giới Tính|    CMND    |Năm sinh|             Địa chỉ            |Số Điện Thoại|");
		System.out.println(
				"    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
		int i = 1;
		for (NhanVien nv : lNhanVien)
			if (nv.sLoaiNV == "QL")
			{
				System.out.printf("    |%3d  |%5s|%23s|%7s  |%12s|%8d|%32s|%13" + "s|\n", i, nv.sMaNV, nv.sTen,
						nv.sGioiTinh, nv.sCMND, nv.iNamSinh, nv.sDiaChi, nv.sSDT);
				i++;
			}
		System.out.println(
				"    ____________________________________________________________________________________________________________________");
		System.out.println();
	}

	public void showThuNgan()
	{
		System.out.println("                            ***DANH SÁCH NHÂN VIÊN THU NGÂN FASHION SHOP***");
		System.out.println(
				"    ____________________________________________________________________________________________________________________");
		System.out.println(
				"    | STT |Mã NV|       Họ và Tên       |Giới Tính|    CMND    |Năm sinh|             Địa chỉ            |Số Điện Thoại|");
		System.out.println(
				"    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
		int i = 1;
		for (NhanVien nv : lNhanVien)
			if (nv.sLoaiNV == "TN")
			{
				System.out.printf("    |%3d  |%5s|%23s|%7s  |%12s|%8d|%32s|%13" + "s|\n", i, nv.sMaNV, nv.sTen,
						nv.sGioiTinh, nv.sCMND, nv.iNamSinh, nv.sDiaChi, nv.sSDT);
				i++;
			}
		System.out.println(
				"    ____________________________________________________________________________________________________________________");
		System.out.println();
	}

	public void showTuVan()
	{
		System.out.println("                         ***DANH SÁCH NHÂN VIÊN TƯ VẤN MUA HÀNG FASHION SHOP***");
		System.out.println(
				"    ____________________________________________________________________________________________________________________");
		System.out.println(
				"    | STT |Mã NV|       Họ và Tên       |Giới Tính|    CMND    |Năm sinh|             Địa chỉ            |Số Điện Thoại|");
		System.out.println(
				"    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
		int i = 1;
		for (NhanVien nv : lNhanVien)
			if (nv.sLoaiNV == "TV")
			{
				System.out.printf("    |%3d  |%5s|%23s|%7s  |%12s|%8d|%32s|%13" + "s|\n", i, nv.sMaNV, nv.sTen,
						nv.sGioiTinh, nv.sCMND, nv.iNamSinh, nv.sDiaChi, nv.sSDT);
				i++;
			}
		System.out.println(
				"    ____________________________________________________________________________________________________________________");
		System.out.println();

	}

	public void addNhanVien()
	{
		Scanner sc = new Scanner(System.in);
		System.out.println("____Hãy nhập thông tin nhân viên mới____");
		System.out.println("____Chọn loại nhân viên: ");
		System.out.println("Quản Lý  : QL");
		System.out.println("Thu Ngân : TN");
		System.out.println("Tư Vấn   : TV");
		String loaiNV = sc.next();
		// System.out.println(lNhanVien.get(2).sMaNV);
		System.out.print("Tên nhân viên: ");
		String ten = sc.next();

		System.out.print("Giới tính: ");
		String gt = sc.next();

		System.out.print("CMND: ");
		String cmnd = sc.next();

		System.out.print("Năm sinh: ");
		int namsinh = Integer.parseInt(sc.next());

		System.out.print("Địa chỉ: ");
		String dc = sc.next();

		System.out.print("SĐT: ");
		String sdt = sc.next();
		sc.nextLine();
		String maNV;
		int so = 0;
		//System.out.println(loaiNV);
		for (int j = lNhanVien.size() - 1; j >= 0; j--)
		{
			if (lNhanVien.get(j).sLoaiNV.equals(loaiNV))
			{
				so = Integer.parseInt(lNhanVien.get(j).sMaNV.substring(2, 4)); //substring(a,b) lấy đoạn chuỗi từ a đến b-1
				break;
			}
		}
		if (so < 9)
			maNV = loaiNV + "0" + String.valueOf(so + 1);
		else
			maNV = loaiNV + String.valueOf(so + 1);
		// sMaNV, sTen, iNamSinh, sSDT, sDiaChi, sCMND, sLoaiNV, sGioiTinh

		int i = 0;

		switch (loaiNV)
		{
		case "QL":
			NhanVien ql = new QuanLy(maNV, ten, namsinh, sdt, dc, cmnd, "QL", gt);
			lNhanVien.add(ql);
			i++;
			break;
		case "TN":
			NhanVien tn = new ThuNgan(maNV, ten, namsinh, sdt, dc, cmnd, "TN", gt);
			lNhanVien.add(tn);
			i++;
			break;
		case "TV":
			NhanVien tv = new TuVan(maNV, ten, namsinh, sdt, dc, cmnd, "TV", gt);
			lNhanVien.add(tv);
			i++;
			break;
		default:
			break;
		}
		if (i > 0)
		{
			System.out.println("Thông tin nhân viên mới đã được lưu !!");
			System.out.println("__________________________________________________________");
		} else
			{
			System.out.println("Có gì đó không ổn, bạn gõ 1 để nhập lại thông tin nhé !!");
			System.out.println("__________________________________________________________");
		}
	}

	public void delNhanVien()
	{
		boolean i = true;
		Scanner sc = new Scanner(System.in);
		System.out.print("Nhập mã nhân viên của nhân viên bạn muốn xóa: ");
		String m = sc.next();
		for (NhanVien nv : lNhanVien)
		{
			if (nv.sMaNV.equals(m))
			{
				lNhanVien.remove(nv);
				System.out.println("Nhân viên " + m + " đã bị xóa !");
				System.out.println("__________________________________________________________");
				i = false;
				break;
			}
		}
		if (i)
		{
			System.out.println("Mã nhân viên này không tồn tại, bạn gõ 3 đã tiếp tục xóa !");
			System.out.println("__________________________________________________________");
		}

	}

	public void editNhanVien()
	{
		boolean i = true;
		Scanner sc = new Scanner(System.in);
		System.out.print("Nhập mã nhân viên của nhân viên bạn muốn chỉnh sửa: ");
		String m = sc.next();
		for (NhanVien nv : lNhanVien)
		{
			if (nv.sMaNV.equals(m))
			{
				System.out.print("Tên nhân viên: ");
				String ten = sc.next();
				nv.setTen(ten);
				System.out.print("Giới tính: ");
				String gt = sc.next();
				nv.setsGioiTinh(gt);
				System.out.print("CMND: ");
				String cmnd = sc.next();
				nv.setCMND(cmnd);
				System.out.print("Năm sinh: ");
				int namsinh = Integer.parseInt(sc.next());
				nv.setNamSinh(namsinh);
				System.out.print("Địa chỉ: ");
				String dc = sc.next();
				nv.setDiaChi(dc);
				System.out.print("SĐT: ");
				String sdt = sc.next();
				nv.setSDT(sdt);
				sc.nextLine();

				System.out.println("Thông tin nhân viên " + m + " đã được cập nhật !");
				System.out.println("__________________________________________________________");
				i = false;
				break;
			}
		}
		if (i) {
			System.out.println("Mã nhân viên này không tồn tại, bạn gõ 2 đã tiếp tục xóa !");
			System.out.println("__________________________________________________________");

		}

	}

	public void searchNhanVien()
	{
		boolean i = true;
		Scanner sc = new Scanner(System.in);
		System.out.println("                                        ***TÌM NHÂN VIÊN***");
		System.out.print("Nhập mã nhân viên của nhân viên bạn muốn tìm: ");
		String m = sc.nextLine();
		System.out.println(
				"    ______________________________________________________________________________________________________________");
		System.out.println(
				"    |Mã NV|       Họ và Tên       |Giới Tính|    CMND    |Năm sinh|             Địa chỉ            |Số Điện Thoại|");
		System.out.println(
				"    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
		for (NhanVien nv : lNhanVien)
		{
			if (nv.getMaNV().equals(m))
			{
				System.out.printf("    |%5s|%23s|%7s  |%12s|%8d|%32s|%13" + "s|\n", nv.sMaNV, nv.sTen, nv.sGioiTinh,
						nv.sCMND, nv.iNamSinh, nv.sDiaChi, nv.sSDT);
				System.out.println(
						"    ______________________________________________________________________________________________________________");
				System.out.println();
				i = false;
				break;
			}
		}
		if (i)
		{
			System.out.println("Mã nhân viên này không tồn tại, bạn gõ 5 đã tiếp tục xóa !");
			System.out.println("__________________________________________________________");
		}

	}
}

File: QUANLYSHOP_C#/QUANLYSHOP/QuanLyNhanVien.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Linq;

namespace QUANLYSHOP
{
    public class QuanLyNhanVien
    {
        private List<NhanVien> lNhanVien;

        public List<NhanVien> LNhanVien { get => lNhanVien; set => lNhanVien = value; }

        protected TuVan ThemTuVan()
        {
            Console.Write("     Mã nhân viên: ");
            String sMaNV = Console.ReadLine();

            int nv = KiemTraTonTaiMaNV(sMaNV);

            while (nv > 0)
            {
                Console.WriteLine("  WARNING: Mã nhân viên đã tồn tại. Vui lòng nhập mã khác ");
                sMaNV = Console.ReadLine();
                nv = KiemTraTonTaiMaNV(sMaNV);

            }


            Console.Write("     Tên nhân viên: ");
            String sTen = Console.ReadLine();

            Console.Write("     Giới tính: ");
            String sGioiTinh = Console.ReadLine();

            Console.Write("     CMND: ");
            String sCMND = Console.ReadLine();

            Console.Write("     Năm sinh: ");
            int iNamSinh = int.Parse(Console.ReadLine());

            Console.Write("     Địa chỉ: ");
            String sDiaChi = (Console.ReadLine());

            Console.Write("     SĐT: ");
            String sSDT = Console.ReadLine();

            TuVan tv = new TuVan(sMaNV, sTen, iNamSinh, sSDT, sDiaChi, sCMND, "TV", sGioiTinh, 0, 0);
            return tv;
        }

        protected ThuNgan ThemThuNgan()
        {
            Console.Write("      Mã nhân viên: ");
            String sMaNV = Console.ReadLine();

            int nv = KiemTraTonTaiMaNV(sMaNV);

            while (nv > 0)
            {
                Console.WriteLine("  WARNING: Mã nhân viên đã tồn tại. Vui lòng nhập mã khác ");
                sMaNV = Console.ReadLine();

                nv = KiemTraTonTaiMaNV(sMaNV);
            }

            Console.Write("     Tên nhân viên: ");
            String sTen = Console.ReadLine();

            Console.Write("     Giới tính: ");
            String sGioiTinh = Console.ReadLine();

            Console.Write("     CMND: ");
            String sCMND = Console.ReadLine();

            Console.Write("     Năm sinh: ");
            int iNamSinh = int.Parse(Console.ReadLine());

            Console.Write("     Địa chỉ: ");
            String sDiaChi = (Console.ReadLine());

            Console.Write("     SĐT: ");
            String sSDT = Console.ReadLine();

            ThuNgan tn = new ThuNgan();
            return tn;
        }
        protected QuanLy ThemQuanLy()
        {
            Console.Write("     Mã nhân viên: ");
            String sMaNV = Console.ReadLine();

            int nv = KiemTraTonTaiMaNV(sMaNV);

            while (nv > 0)
            {
                Console.Write("  WARNING: Mã nhân viên đã tồn tại. Vui lòng nhập mã khác ");
                sMaNV = Console.ReadLine();
                nv = KiemTraTonTaiMaNV(sMaNV);
            }

            Console.Write("     Tên nhân viên: ");
            String sTen = Console.ReadLine();

            Console.Write("     Giới tính: ");
            String sGioiTinh = Console.ReadLine();

            Console.Write("     CMND: ");
            String sCMND = Console.ReadLine();

            Console.Write("     Năm sinh: ");
            int iNamSinh = Convert.ToInt16(Console.ReadLine());

            Console.Write("     Địa chỉ: ");
            String sDiaChi = (Console.ReadLine());

            Console.Write("     SĐT: ");
            String sSDT = Console.ReadLine();

            QuanLy quanLy = new QuanLy(sMaNV, sTen, iNamSinh, sSDT, sDiaChi, sCMND, "NV", sGioiTinh);
            return quanLy;
        }

        private int KiemTraTonTaiMaNV(string manv)
        {
            return (from nhanvien in lNhanVien
                    where nhanvien.SMaNV == manv
                    select nhanvien).Count();
        }

        private String TaoMaNV(String sLoaiNV)
        {
            int so = 0;
            for (int j = this.LNhanVien.Count - 1; j >= 0; j--)
            {
                if (LNhanVien[j].Equals(sLoaiNV))
                {
                    so = int.Parse(this.LNhanVien[j].SMaNV.Substring(2, 4)); //substring(a,b) lấy đoạn chuỗi từ a đến b-1
                    break;
                }
            }

            if (so < 9)
                return sLoaiNV + "0" + 1;
            else
                return sLoaiNV + (so + 1);
        }
        public void addNhanVien()
        {
            Console.WriteLine("________________________________________________");
            Console.WriteLine("              THÔNG TIN NHÂN VIÊN MỚI");
            Console.WriteLine("         ---VUI LÒNG CHỌN LOẠI NHÂN VIÊN--- ");

            String sLoaiNV = "";
            do
            {
                Console.WriteLine("     QUẢN LÝ       : QL");
                Console.WriteLine("     THU NGÂN      : TN");
                Console.WriteLine("     TƯ VẤN VIÊN   : TV");
                Console.Write("\n LỰA CHỌN CỦA BẠN LÀ: ");
                sLoaiNV = Console.ReadLine();
            }
            while (!sLoaiNV.Equals("QL") && !sLoaiNV.Equals("TN") && !sLoaiNV.Equals("TV"));

            String maNV = TaoMaNV(sLoaiNV);

            // sMaNV, sTen, iNamSinh, sSDT, sDiaChi, sCMND, sLoaiNV, sGioiTinh

            switch (sLoaiNV)
            {
                case "QL":
                    QuanLy ql = ThemQuanLy();
                    LNhanVien.Add(ql);
                    break;

                case "TN":
                    ThuNgan tn = ThemThuNgan();
                    LNhanVien.Add(tn);
                    break;
                case "TV":
                    TuVan tv = ThemTuVan();
                    LNhanVien.Add(tv);
                    break;
                default:
                    break;
            }

            Console.WriteLine("     Thông tin nhân viên mới đã được lưu !!");
            Console.WriteLine("________________________________________________________");
        }
        public QuanLyNhanVien()
        {
            lNhanVien = new List<NhanVien>();

            // quản lý
            NhanVien ql1 = new QuanLy("QL01", "Trần Thị Thanh Tâm", 1999, "0988188449", "Quận 9, Tp.HCM", "251274983", "QL", "Nữ");
            lNhanVien.Add(ql1);
            NhanVien ql2 = new QuanLy("QL02", "Nguyễn Trần Tiểu Vy", 1999, "0988188733", "Quận 8, Tp.HCM", "251287483", "QL", "Nam");
            lNhanVien.Add(ql2);

            // thu ngân
            NhanVien tn1 = new ThuNgan("TN01", "Lê Quỳnh Trâm Anh", 1998, "0986978673", "Quận Thủ Đức, Tp.HCM", "251274453", "TN", "Nữ", 0, 0);
            lNhanVien.Add(tn1);
            NhanVien tn2 = new ThuNgan("TN02", "Nguyễn Đình Thái Ngân", 1997, "0984567673", "Quận Bình Thạnh, Tp.HCM", "256784453", "TN", "Nam", 0, 0);
            lNhanVien.Add(tn2);

            // tư vấn
            NhanVien tv1 = new TuVan("TV01", "Cao Mỹ Nhân", 2001, "0987983245", "Quận 2, Tp.HCM", "245274983", "TV", "Nữ", 0, 0);
            lNhanVien.Add(tv1);
            NhanVien tv2 = new TuVan("TV02", "Trần Nguyễn Tuấn Anh", 2001, "0935983245", "Quận 3, Tp.HCM", "246274983", "TV", "Nam", 0, 0);
            lNhanVien.Add(tv2);
        }

        public void showNhanVien()
        {

            Console.WriteLine("____Chọn danh sách: ");
            Console.WriteLine("	    0.Toàn bộ nhân viên");
            Console.WriteLine("	    1.Quản Lý");
            Console.WriteLine("	    2.Thu Ngân");
            Console.WriteLine("	    3.Tư Vấn Viên");
            Console.Write("\n LỰA CHỌN CỦA BẠN LÀ: ");
            int choose = int.Parse(Console.ReadLine());

            switch (choose)
            {
                case 0:
                    showAllNhanVien();
                    break;
                case 1:
                    showQuanLy();
                    break;
                case 2:
                    showThuNgan();
                    break;
                case 3:
                    showTuVan();
                    break;
                default:
                    break;
            }
        }

        public void showAllNhanVien()
        {
            Console.WriteLine("                                                 DANH SÁCH TẤT CẢ NHÂN VIÊN FASHION'S SHOP");
            Console.WriteLine("    __________________________________________________________________________________________________________________________\n");
            Console.WriteLine("    |STT  |Mã NV|       Họ và Tên          |Giới Tính|      CMND       |Năm sinh|            Địa chỉ           |Số Điện Thoại|");
            Console.WriteLine("    __________________________________________________________________________________________________________________________");
            int i = 1;

            foreach (var nv in LNhanVien)
            {
                // In thông tin của nhân viên ra màn hình
                Console.WriteLine("      {0,-5}{1,-10}{2,-25}{3,-11}{4,-16}{5,-14}{6,-28}{7,-10} ", i, nv.SMaNV, nv.STen, nv.SGioiTinh, nv.SCMND, nv.INamSinh, nv.SDiaChi, nv.SSDT);
                i++;
            }
            Console.WriteLine("    __________________________________________________________________________________________________________________________");

            Console.WriteLine();
        }

        public void showQuanLy()
        {
            Console.WriteLine("                            DANH SÁCH NHÂN VIÊN QUẢN LÝ FASHION'S SHOP");
            Console.WriteLine(
                    "    ____________________________________________________________________________________________________________________");
            Console.WriteLine(
                    "    | STT |Mã NV|       Họ và Tên       |Giới Tính|    CMND    |Năm sinh|             Địa chỉ            |Số Điện Thoại|");
            Console.WriteLine(
                    "    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
            int i = 1;
            foreach (var nv in LNhanVien)
                if (nv.SLoaiNV == "QL")
                {
                    Console.Write("     {0,-5}", i++);
                    nv.ShowInfo();

                }
            Console.WriteLine(
                    "    ____________________________________________________________________________________________________________________");
            Console.WriteLine();
        }

        public void showThuNgan()
        {
            Console.WriteLine("                            DANH SÁCH NHÂN VIÊN THU NGÂN FASHION'S SHOP");
            Console.WriteLine(
                    "    ____________________________________________________________________________________________________________________");
            Console.WriteLine(
                    "    | STT |Mã NV|       Họ và Tên       |Giới Tính|    CMND    |Năm sinh|             Địa chỉ            |Số Điện Thoại|");
            Console.WriteLine(
                    "    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
            int i = 1;
            foreach (var nv in LNhanVien)
                if (nv.SLoaiNV == "TN")
                {
                    Console.Write("     {0,-5}", i++);
                    nv.ShowInfo();
                }
            Console.WriteLine(
                    "    ____________________________________________________________________________________________________________________");
            Console.WriteLine();
        }

        public void showTuVan()
        {
            Console.WriteLine("                         DANH SÁCH NHÂN VIÊN TƯ VẤN FASHION'S SHOP");
            Console.WriteLine(
                    "    ____________________________________________________________________________________________________________________");
            Console.WriteLine(
                    "    | STT |Mã NV|       Họ và Tên       |Giới Tính|    CMND    |Năm sinh|             Địa chỉ            |Số Điện Thoại|");
            Console.WriteLine(
                    "    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
            int i = 1;
            foreach (var nv in LNhanVien)
                if (nv.SLoaiNV == "TV")
                {
                    Console.Write("     {0,-5}", i++);
                    nv.ShowInfo();
                }
            Console.WriteLine(
                    "    ____________________________________________________________________________________________________________________");
            Console.WriteLine();

        }


        public void delNhanVien()
        {

            Console.WriteLine("     Nhập mã nhân viên của nhân viên bạn muốn xóa: ");
            string m = Console.ReadLine();
            var nhanvien = (from NV in lNhanVien
                            where NV.SMaNV == m
                            select NV).Count();
            if (nhanvien == 0)
            {
                Console.WriteLine("     Mã nhân viên này không tồn tại, bạn gõ 3 đã tiếp tục  !");
                Console.WriteLine("__________________________________________________________");

            }
            else if (nhanvien > 0)
            {
                var nv = (from NV in lNhanVien
                          where NV.SMaNV == m
                          select NV).First();
                lNhanVien.Remove(nv);
                Console.WriteLine("     Nhân viên " + m + " đã bị xóa !");
                Console.WriteLine("__________________________________________________________");
            }

        }


        public void editNhanVien()
        {

            Console.WriteLine("     Nhập mã nhân viên của nhân viên bạn muốn chỉnh sửa: ");
            string maNV = Console.ReadLine();

            var nhanvien = KiemTraTonTaiMaNV(maNV);

            if (nhanvien == 0)
            {
                Console.WriteLine("     Mã nhân viên này không tồn tại, bạn gõ 2 đã tiếp tục  !");
                Console.WriteLine("__________________________________________________________");

            }

            else if (nhanvien > 0)
            {
                var nv = (from NV in lNhanVien
                          where NV.SMaNV == maNV
                          select NV).First();
                Console.Write("     Tên nhân viên: ");
                String ten = Console.ReadLine();
                nv.STen = ten;
                Console.Write("     Giới tính: ");
                String gt = Console.ReadLine();
                nv.SGioiTinh = gt;

                Console.Write("     CMND: ");
                String cmnd = Console.ReadLine();
                nv.SCMND = (cmnd);

                Console.Write("     Năm sinh: ");
                int namsinh = int.Parse(Console.ReadLine());
                nv.INamSinh = (namsinh);

                Console.Write("     Địa chỉ: ");
                String dc = Console.ReadLine();
                nv.SDiaChi = (dc);

                Console.Write("     SĐT: ");
                String sdt = Console.ReadLine();
                nv.SSDT = (sdt);


                Console.WriteLine("     Thông tin nhân viên " + maNV + " đã được cập nhật !");
                Console.WriteLine("__________________________________________________________");
            }

        }

        public void searchNhanVien()
        {

            Console.WriteLine("                   ====TÌM NHÂN VIÊN====");
            Console.WriteLine("     Nhập mã nhân viên của nhân viên bạn muốn tìm: ");
            Console.WriteLine();
            string m = Console.ReadLine();


            var search = (from nv in lNhanVien
                          where nv.SMaNV == m
                          select nv).Count();

            if (search == 0)
            {
                Console.WriteLine();
                Console.WriteLine("     Mã nhân viên này không tồn tại, bạn gõ 5 đã tiếp tục TÌM !");
                Console.WriteLine("_____________________________________________________________________\n");
            }

            else
            {
                var nhanVien = (from nv in lNhanVien
                                where nv.SMaNV.Contains(m) == true
                                select nv).First();
                Console.WriteLine(
                    "    ______________________________________________________________________________________________________________");
                Console.WriteLine(
                        "    |Mã NV|       Họ và Tên       |Giới Tính|    CMND    |Năm sinh|             Địa chỉ            |Số Điện Thoại|");
                Console.WriteLine(
                        "    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");

                Console.WriteLine("    {0,-10}{1,-25}{2,-9}{3,-12}{4,-15}{5,-28}{6,-20} ", nhanVien.SMaNV, nhanVien.STen, nhanVien.SGioiTinh, nhanVien.SCMND, nhanVien.INamSinh, nhanVien.SDiaChi, nhanVien.SSDT);
                Console.WriteLine("    ______________________________________________________________________________________________________________");
                Console.WriteLine();
            }
        }
        public void Tinhluong()
        {
            Console.WriteLine("     NHẬP MÃ NHÂN VIÊN BẠN CẦN TÍNH LƯƠNG: TV01");

            //string m = Console.ReadLine();

            //var search = (from nv in lNhanVien
            //              where nv.SMaNV == m
            //              select nv).Count();
            //if (search == 0)
            //{
            //    Console.WriteLine();
            //    Console.WriteLine("     Mã nhân viên này không tồn tại");
            //    Console.WriteLine("_____________________________________________________________________\n");
            //}

            //else
            //{
            //    var nhanVien = (from nv in lNhanVien
            //                    where nv.SMaNV.Contains(m) == true
            //                    select nv).First();

            //}
            Console.WriteLine("     NHẬP SỐ NGÀY ĐI LÀM: 28");
            Console.WriteLine("     NHẬP SỐ GIỜ TĂNG CA: 10");
            Console.WriteLine("____________________________________________\n");
            Console.WriteLine("                  BẢNG LƯƠNG\n");
            Console.WriteLine("     Tên nhân viên : Cao Mỹ Nhân");
            Console.WriteLine("     Tổng lương tháng này: 7250000 VNĐ");
        }
    }
}


